cranelift/codegen/src/isa/riscv64/inst/emit.rs

```diff
@@ -2893,6 +2893,38 @@ impl MachInstEmit for Inst {
                 .emit(&[], sink, emit_info, state);
                 sink.bind_label(label_done);
             }
+            &Inst::ElfTlsGetAddr { ref symbol, rd } => {
+                let rd = allocs.next_writable(rd);
+                assert_eq!(a0(), rd.to_reg());
+                sink.add_reloc(Reloc::RiscvTlsGd, symbol, 0);
+                Inst::Auipc {
+                    rd: rd,
+                    imm: Imm20::from_bits(0),
+                }
+                .emit(&[], sink, emit_info, state);
+                Inst::AluRRImm12 {
+                    alu_op: AluOPRRI::Addi,
+                    rd: rd,
+                    rs: rd.to_reg(),
+                    imm12: Imm12::from_bits(0),
+                }
+                .emit(&[], sink, emit_info, state);
```

Comment:
The relocations here don't seem to be correct. You first need an `auipc` with `R_RISCV_TLS_GD_HI20` and then an `addi` with `R_RISCV_PCREL_LO12_I`. You only have the first one.

Reply:
ok.